### ISSUE TITLE
Speed up key-generation

### ISSUE THREAD

**Issue by Firesphere** [StarshipEngineer/OpenVPN-Setup#41]
To speed up key generation, run random commands like `tree /` and alike. These don't alter the system, but do generate entropy.

**Comment by Firesphere** [comment]
Another option is to run `sudo dd if=/dev/mmcblk0p2 of=/dev/null`

But don't mix the if and of values. Or you'll writing /dev/null to your card!

**Comment by 0-kaladin** [comment]
I looked into this for pivpn.io but after trying a bunch of different things that speed up key generation for pgp, nothing speed up the keys on the raspberry pi for building the openvpn key.
do you have a command/script/example of this working to speed up the keygen for openvpn?  or maybe on the raspberry pi the throttling factor isn't a lack of random data but rather just the cpu.  I think you are assuming it goes slow due to lack of entropy in the rand pool but I haven't seen that to be the case yet.  I think the limiting factor is just the CPU.  Also when I run this on an amazon server as opposed to a raspberry pi it goes 5x quicker.

**Comment by PenguinDUB** [comment]
I agree on the CPU being a limiting factor for the initial openVPN keys. My Pi3 has an "ARM Cortex-A53 processing cores running at 1.2GHz with 32kB Level 1 and 512kB Level 2 cache memory" which is impressive for such a robust, little device but not so hot compared other ARM or x86 chips. I like Firesphere's approach for entropy.
